easybuild/easyconfigs/n/NLopt/NLopt-2.6.1-GCCcore-8.2.0.eb

```diff
@@ -0,0 +1,38 @@
+# This file is an EasyBuild reciPY as per https://github.com/easybuilders/easybuild
+# Author: Pablo Escobar Lopez
+# Swiss Institute of Bioinformatics
+# Biozentrum - University of Basel
+
+easyblock = 'CMakeMake'
+
+name = 'NLopt'
+version = '2.6.1'
+
+homepage = 'http://ab-initio.mit.edu/wiki/index.php/NLopt'
+description = """ NLopt is a free/open-source library for nonlinear optimization, 
+ providing a common interface for a number of different free optimization routines 
+ available online as well as original implementations of various other algorithms. """
+
+toolchain = {'name': 'GCCcore', 'version': '8.2.0'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/stevengj/nlopt/archive']
+sources = [{
+    'download_filename': 'v%(version)s.tar.gz',
+    'filename': SOURCE_TAR_GZ
+}]
+checksums = ['66d63a505187fb6f98642703bd0ef006fedcae2f9a6d1efa4f362ea919a02650']
+
+builddependencies = [
+    ('CMake', '3.13.3'),
+    ('binutils', '2.31.1'),
+]
+
+configopts = '--enable-shared --without-matlab --without-octave'
+
+sanity_check_paths = {
+    'files': ['lib/libnlopt.%s' % SHLIB_EXT, 'include/nlopt.h'],
```

Comment:
We have `libnlopt.a` listed here before as well, but now it seems it's no longer being installed, probably a side-effect of the switch to CMake? Can we get it back easily, for example by specifying the right option to the `cmake` command?

Reply:
NLopt uses the LGPL license, anything that is linked to it becomes has to follow the Gnu LGL license. The CMake file only allows for static or shared; not both.  I would have to rewrite the CMake to allow building both.

Reply:
Not really, you can actually tell EasyBuild to perform the installation in two iterations: once to install the static libraries, once to install the shared libraries.
That can be done easily simply by making `configopts` a list of string values (rather than a single string value).

But, the current `configopts` don't make much sense, those are not CMake options?!

Reply:
I was going to ask if there is a way to build something twice.  configopts list works great, both static and shared are built now.  And I removed the old configopts which did nothing.

Reply:
Please enhance the sanity check to also check for `libnlopt.a`?